File: WaterandJugProblem2.py
class Solution:
    def canMeasureWater(self, x, y, z):
        """
        :type x: int
        :type y: int
        :type z: int
        :rtype: bool
        """
        if z == 0 or z == x + y:
            return True
        if z > x + y:
            return False

        if x > y:
            x, y = y, x

        v = 0
        while True:
            if v < x:
                v += y
            else:
                v -= x
            if v == z:
                return True
            if v == 0:
                return False

        return False


if __name__ == '__main__':
    x, y, z = (int(num) for num in input().split())
    solution = Solution()
    print(solution.canMeasureWater(x, y, z))

File: WaterandJugProblem2.cpp
#include<iostream>

using namespace std;

class Solution {
public:
    bool canMeasureWater(int x, int y, int z) {
    	
    	if(z == 0) return true;
    	if(z == x + y) return true;
    	
        if(z > x + y) return false;
        
        if(x > y)  swap(x, y);
        int v = 0;
        
        do {
        	if(v < x) v += y;
        	else v -= x;
        	if(v == z) return true;
		} while(v != 0);
		
		return false;
        
        
    }
    
};

int main() {
	
	int x, y, z;
	cin>>x>>y>>z;
	
	Solution *solution = new Solution();
	solution->canMeasureWater(x, y, z) ? cout<<"true" : cout<<"false";
	
	return 0;
} 
